server/stt/stt_info_porssi.py

```diff
@@ -0,0 +1,155 @@
+
+NEWSML_NS = "http://iptc.org/std/NewsML/2003-10-10/"
+
+
+class STTInfoPorssi(NewsMLOneFeedParser):
+    """STT Info Porssi XML ingest.
+
+    STT Info Porssi is a custom NewsML 1.2 format
+    """
+
+    NAME = "stt_info_porssi"
+
+    label = "STT Info Porssi"
+
+    def __init__(self):
+        super().__init__()
+        self.default_mapping = {
+            "guid": {
+                "xpath": "Identification/NewsIdentifier/NewsItemId/text()",
+                "filter": lambda i: "stt-info-porssi_{}".format(i),
+            },
+            "headline": "NewsComponent/NewsLines/HeadLine",
+            "slugline": "NewsComponent/AdministrativeMetadata/Source/Party/@FormalName",
+            "body_html": self.get_body,
+            "name": {
+                "xpath": "NewsComponent/AdministrativeMetadata/Source/Party/@FormalName",
+                "key_hook": lambda item, name: item.setdefault("extra", {}).__setitem__(
+                    "ntb_pub_name", name
+                ),
+            },
+        }
+
+    def can_parse(self, xml):
+        return (
+            xml.tag == "{http://iptc.org/std/NewsML/2003-10-10/}NewsML"
+            and xml.get("Version", "") == "1.2"
+        )
+
+    def parse(self, xml, provider=None):
+        try:
+            # Remove namespace for easier XPath
+            xml = deepcopy(xml)
+            for elt in xml.iter():
+                elt.tag = elt.tag.replace("{" + NEWSML_NS + "}", "")
+            news_items = xml.findall("NewsItem")
+
+            selected = None
+            for news_item in news_items:
+                try:
+                    lang = news_item.xpath(
+                        "NewsComponent/DescriptiveMetadata/Language/@FormalName"
+                    )[0]
+                except IndexError:
+                    logger.warning(
+                        "missing language in item, ignoring it.\nxml: {xml}".format(
+                            xml=etree.tostring(news_item, encoding="unicode")
+                        )
+                    )
+                    continue
+
+                if selected is None or lang in ("fi", "sv", "en"):
+                    selected = news_item
+                if lang == "fi":
+                    break
+
+            if selected is None:
+                raise ParserError.parseFileError(
+                    source=etree.tostring(xml, encoding="unicode")
+                )
+
+            guid = selected.findtext("Identification/NewsIdentifier/NewsItemId")
+            body = self.get_body(selected)
+
+            # Use xpath() for attribute selectors
+            source_elements = selected.xpath(
+                "NewsComponent/AdministrativeMetadata/Source/Party/@FormalName"
+            )
+            source = source_elements[0] if source_elements else "STT"
+
+            headline = selected.findtext("NewsComponent/NewsLines/HeadLine") or ""
+
+            item = {
+                "guid": f"stt-info-porssi_{guid}",
```

Comment:
`urn:` might be a better prefix for it, but you could also use that `PublicIdentifier` value as is